src/sentry/sentry_apps/external_requests/select_requester.py

```diff
@@ -78,8 +78,9 @@ def run(self) -> SelectRequesterResult:
                 message = "select-requester.request-failed"
 
             logger.info(message, extra=extra)
-            raise APIError(
-                f"Something went wrong while getting SelectFields from {self.sentry_app.slug}"
+            raise SentryAppIntegratorError(
+                message=f"Something went wrong while getting SelectFields from {self.sentry_app.slug}",
+                extras={"webhook_context": {"error_type": message, **extra}},
```

Comment:
Does it make sense to include the extras here & is this something a user will see in their network requests pane?

Reply:
Because these extras are under the `"webhook_context"` key this won't be publicly seen by end users (or in the network requests). If this dict was under the `"public_context"` key then yes these would be in the response body and visible in the network requests pane

Reply:
Would it make sense to separate public from "private" context data in separate dictionaries as part of the two `SentryAppError` types? It would make the purpose of each more explicit in code I think.

Reply:
sure that makes sense, it's a bit nested rn